gc.c

```diff
@@ -11300,7 +11370,9 @@ gc_stat_internal(VALUE hash_or_sym)
         rb_hash_aset(hash, gc_stat_symbols[gc_stat_sym_##name], SIZET2NUM(attr));
 
     SET(count, objspace->profile.count);
-    SET(time, (size_t) (objspace->profile.total_time_ns / (1000 * 1000) /* ns -> ms */)); // TODO: UINT64T2NUM
+    SET(time, (size_t)ns_to_ms(objspace->profile.total_time_ns)); // TODO: UINT64T2NUM
+    SET(marking_time, (size_t)ns_to_ms(objspace->profile.marking_time_ns));
+    SET(sweeping_time, (size_t)ns_to_ms(objspace->profile.sweeping_time_ns));
```

Comment:
Hum, I haven't looked at the patch in close detail yet, but couldn't `profile.total_time_ns` be eliminated?

Intuitively I'd think we could compute it on the fly with `marking_time_ns + sweeping_time_ns`? That would be one less counter to increment.

Reply:
`total_time_ns` also includes time spent on finalizers, but I guess if we added a `finalizing_time_ns` then `total_time_ns == marking_time_ns + sweeping_time_ns + finalizing_time_ns`.

Reply:
Couldn't we consider `finalizing` time as part of sweeping?

Reply:
Oh never mind, it looks like `gc_enter_event_measure_p` returns false for finalizing, so the time doesn't get measured. So you're right, `total_time_ns == marking_time_ns + sweeping_time_ns`.